### ISSUE TITLE
Error on startup after 12.1.1 upgrade

### ISSUE THREAD

**Issue by dpedrovich** [umbraco/Umbraco-CMS#14693]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

12.1.1

### Bug summary

After upgrading a website from v12.0.1 to v12.1.1 it throws the following error:

`FileNotFoundException: Could not load file or assembly 'Humanizer, Version=2.14.0.0, Culture=neutral, PublicKeyToken=979442b78dfc278e, processorArchitecture=MSIL'. The system cannot find the file specified.
System.Reflection.RuntimeAssembly.InternalLoad(AssemblyName assemblyName, ref StackCrawlMark stackMark, AssemblyLoadContext assemblyLoadContext, RuntimeAssembly requestingAssembly, bool throwOnFileNotFound)
System.Reflection.Assembly.Load(AssemblyName assemblyRef)
Umbraco.Cms.Core.Composing.ReferenceResolver.ResolveAssemblies()
Umbraco.Cms.Core.Composing.FindAssembliesWithReferencesTo.Find()
Umbraco.Cms.Core.Composing.DefaultUmbracoAssemblyProvider.get_Assemblies()
Umbraco.Cms.Core.Composing.TypeFinder.GetAllAssemblies()
Umbraco.Cms.Core.Composing.TypeFinder.GetFilteredAssemblies(IEnumerable<Assembly> excludeFromResults, string[] exclusionFilter)
Umbraco.Cms.Core.Composing.TypeFinder.get_AssembliesToScan()
Umbraco.Cms.Core.Composing.TypeLoader.get_AssembliesToScan()
Umbraco.Cms.Core.Composing.TypeLoader+<>c__DisplayClass21_0<T>.<GetTypes>b__0()
Umbraco.Cms.Core.Composing.TypeLoader.GetTypesInternalLocked(Type baseType, Type attributeType, Func<IEnumerable<Type>> finder, string action, bool cache)
Umbraco.Cms.Core.Composing.TypeLoader.GetTypesInternal(Type baseType, Type attributeType, Func<IEnumerable<Type>> finder, string action, bool cache)
Umbraco.Cms.Core.Composing.TypeLoader.GetTypes<T>(bool cache, IEnumerable<Assembly> specificAssemblies)
Umbraco.Cms.Core.DependencyInjection.UmbracoBuilderExtensions.AddAllCoreCollectionBuilders(IUmbracoBuilder builder)
Umbraco.Cms.Core.DependencyInjection.UmbracoBuilder.AddCoreServices()
Umbraco.Cms.Core.DependencyInjection.UmbracoBuilder..ctor(IServiceCollection services, IConfiguration config, TypeLoader typeLoader, ILoggerFactory loggerFactory, IProfiler profiler, AppCaches appCaches, IHostingEnvironment hostingEnvironment)
Umbraco.Extensions.UmbracoBuilderExtensions.AddUmbraco(IServiceCollection services, IWebHostEnvironment webHostEnvironment, IConfiguration config)
TestProject.Website.Startup.ConfigureServices(IServiceCollection services) in Startup.cs
System.RuntimeMethodHandle.InvokeMethod(object target, Void** arguments, Signature sig, bool isConstructor)
System.Reflection.MethodInvoker.Invoke(object obj, IntPtr* args, BindingFlags invokeAttr)
System.Reflection.RuntimeMethodInfo.Invoke(object obj, BindingFlags invokeAttr, Binder binder, object[] parameters, CultureInfo culture)
Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.InvokeCore(object instance, IServiceCollection services)
Microsoft.AspNetCore.Hosting.GenericWebHostBuilder.UseStartup(Type startupType, HostBuilderContext context, IServiceCollection services, object instance)
Microsoft.AspNetCore.Hosting.GenericWebHostBuilder+<>c__DisplayClass14_0.<UseStartup>b__0(HostBuilderContext context, IServiceCollection services)
Microsoft.Extensions.Hosting.HostBuilder.InitializeServiceProvider()
Microsoft.Extensions.Hosting.HostBuilder.Build()
Umbraco.Cms.Web.Common.Hosting.UmbracoHostBuilderDecorator.Build()
TestProject.Website.Program.Main(string[] args) in Program.cs
`

However, it's only on the production server the error occurs - locally it works fine.
Any ideas on why this is happening? I don't see any references to 'Humanizer' if I search in the project.

I thought it might be related to https://github.com/umbraco/Umbraco-CMS/issues/14658 because of my Startup.cs were also missing the `.AddDeliveryApi()` method. However, after adding it and rebuilding/publishing the website, the error is still present.

### Steps to reproduce

Upgrade a website from v12.0.1 to v12.1.1 in Visual Studio with Nuget, and then attempt to run the site both locally and after being published to a server.

**Comment by elit0451** [comment]
Hi @dpedrovich 👋 

I am not able to reproduce the error unfortunately on a fresh install of v12.0.1 and then update to v12.1.1. Any other specifics you can provide or perhaps can you confirm the same error happens on a fresh install?

**Comment by dpedrovich** [comment]
Hi @elit0451 

I just tried again on a fresh install and the error did not show.
That led me to think that something might be cached on the dev server for my specific project, so I removed everything (except media folder) and did a fresh publish - and boom, the error was gone here as well! 

So apparently, clearing the "umbraco/cache" wasn't enough. I'll keep that in mind for updates in the future :)
